netcore/System.Private.CoreLib/src/System.Runtime.InteropServices/NativeLibrary.cs

```diff
@@ -4,25 +4,36 @@
 
 using System.IO;
 using System.Reflection;
+using System.Runtime.CompilerServices;
 
 namespace System.Runtime.InteropServices
 {
 	partial class NativeLibrary
 	{
-		static IntPtr LoadLibraryByName (string libraryName, Assembly assembly, DllImportSearchPath? searchPath, bool throwOnError) => throw new NotImplementedException ();
-
-		static IntPtr LoadFromPath (string libraryName, bool throwOnError) {
-			// TODO: implement
-			if (throwOnError) {
-				throw new DllNotFoundException();
-			}
-			return IntPtr.Zero;
+		static IntPtr LoadLibraryByName (string libraryName, Assembly assembly, DllImportSearchPath? searchPath, bool throwOnError)
+		{
+			return LoadByName (libraryName,
+			                   (RuntimeAssembly)assembly,
+			                   searchPath.HasValue,
```

Comment:
Is the nullable value decomposition needed here?

Reply:
Believe so, but some of the semantics around nullability aren't clear to me. It's that way in CoreCLR fwiw.